src/java/org/apache/cassandra/db/compaction/PendingRepairManager.java

```diff
@@ -418,21 +428,64 @@ UUID getSessionID()
             return sessionID;
         }
 
+        private boolean isTransientReplica()
```

Comment:
I don't exactly understand this. Can't a session have tables that are transient and not transient? Seems to me like this means that we have to have a session per range so that the session can be either entire transient or not transient at a node.

What this also makes me think is that the cost of a repair session is kind of high, because when you think about it we have to anti-compact per repair session right?

With 6696 and a number of directories matching the number of vnodes maybe anti-compaction becomes basically a free operation.